opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchQueryRequest.java

```diff
@@ -94,11 +97,16 @@ public OpenSearchQueryRequest(IndexName indexName, SearchSourceBuilder sourceBui
   @Override
   public OpenSearchResponse search(Function<SearchRequest, SearchResponse> searchAction,
                                    Function<SearchScrollRequest, SearchResponse> scrollAction) {
+    FetchSourceContext fetchSource = this.sourceBuilder.fetchSource();
+    List<String> includes = fetchSource != null && fetchSource.includes() != null
+            ? Arrays.asList(fetchSource.includes())
+            : List.of();
```

Comment:
May be do this in the query builder?

Reply:
I guess that would reduce the duplicate code

Reply:
OpenSearchQueryRequest and OpenSearchScrollRequest needs a refactor before this makes any sense.

Reply:
Please assist in creating an issue to monitor and manage future modifications.